manager/allocator/networkallocator/networkallocator.go

```diff
@@ -878,3 +890,16 @@ func serviceNetworks(s *api.Service) []*api.NetworkAttachmentConfig {
 	}
 	return s.Spec.Task.Networks
 }
+
+//IsVipOnIngressNetwork check if the vip is in ingress network
+func (na *NetworkAllocator) IsVipOnIngressNetwork(vip *api.Endpoint_VirtualIP) bool {
+	if vip == nil {
+		return false
+	}
+
+	localNet := na.getNetwork(vip.NetworkID)
+	if localNet != nil && localNet.nw != nil {
+		return localNet.nw.Spec.Ingress
```

Comment:
This needs to use `IsIngressNetwork` instead of checking the ingress flag directly. The ingress flag is a relatively recent addition, and ingress networks created with older versions won't have this flag.

Unfortunately `IsIngressNetwork` is defined in `github.com/docker/swarmkit/manager/allocator`, and importing that package here would create an import cycle. It should probably be moved to this package.